### PR TITLE
Android: Fix a possible crash in the file browser.

### PR DESCRIPTION
**Pull request by sigmabeta**
Basically just needed to ignore extensionless files. Whoops.

### CODE REVIEW

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/FileListItem.java [hunk 1]
```diff
@@ -29,7 +30,17 @@ public FileListItem(File file)
...
 		else
 		{
-			String fileExtension = mPath.substring(mPath.lastIndexOf('.'));
+			String fileExtension = null;
+
+			// If the file has no extension, this would throw an exception.
+			try
+			{
+				fileExtension = mPath.substring(mPath.lastIndexOf('.'));
+			} catch (Exception ignored)
```
Comment (Sonicadvance1) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/FileListItem.java]: catch on next line

Comment (sigmabeta) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/FileListItem.java]: Added this rule to code-style-java.jar.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/FileListItem.java [hunk 2]
```diff
@@ -29,7 +30,18 @@ public FileListItem(File file)
...
+			// If the file has no extension, this would throw an exception.
+			try
+			{
+				fileExtension = mPath.substring(mPath.lastIndexOf('.'));
+			}
+			catch (Exception ignored)
+			{
+				// We don't care about files without extensions anyway.
+				mType = TYPE_OTHER;
+			}
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/FileListItem.java]: You can eliminate the use of the try-catch, and exception throwing entirely if you check if lastIndexOf returns -1 or not.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/FileListItem.java [hunk 3]
```diff
@@ -29,23 +30,32 @@ public FileListItem(File file)
...
-			String fileExtension = mPath.substring(mPath.lastIndexOf('.'));
-
-			// Extensions to filter by.
-			Set<String> allowedExtensions = new HashSet<String>(Arrays.asList(".dff", ".dol", ".elf", ".gcm", ".gcz", ".iso", ".wad", ".wbfs"));
+			String fileExtension = null;
 
-			// Check that the file has an appropriate extension before trying to read out of it.
-			if (allowedExtensions.contains(fileExtension))
+			int extensionStart = mPath.lastIndexOf('.');
+			if ((extensionStart < 1))
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/FileListItem.java]: double parentheses aren't necessary

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/FileListItem.java (hunk 2 above)
Comment (sigmabeta) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/FileListItem.java]: I had just copied over the functionality from the old code, in `GameListFragment.fill()`, but this is a better way of doing it.

No point in changing it elsewhere though; the other places where this pattern is used will all be removed eventually.

**Pull request merged**